vendor/engines/secure_rooms/app/models/secure_rooms/occupancy.rb

```diff
@@ -5,11 +5,15 @@ class Occupancy < ActiveRecord::Base
     belongs_to :secure_room, foreign_key: :product_id
     belongs_to :user
     belongs_to :account
+    belongs_to :order_detail
     belongs_to :entry_event, class_name: SecureRooms::Event
     belongs_to :exit_event, class_name: SecureRooms::Event
 
     validates :secure_room, :user, presence: true
 
+    delegate :facility, to: :secure_room
+    delegate :order, to: :order_detail
```

Comment:
I'm not positive, but I think we'll need `allow_nil: true` on this to handle a non-order occupancies. And is this delegation even being used?